Internal/Debug App/Sources/View Controllers/MerchantHeadlessViewController.swift

```diff
@@ -0,0 +1,274 @@
+//
+//  MerchantHeadlessViewController.swift
+//  Debug App
+//
+//  Created by Boris on 20.6.23..
+//  Copyright © 2023 Primer API Ltd. All rights reserved.
+//
+
+import UIKit
+import PrimerSDK
+
+class MerchantHeadlessViewController: UIViewController, PrimerHeadlessUniversalCheckoutDelegate {
+    
+    var settings: PrimerSettings!
+    var clientSession: ClientSessionRequestBody?
+    var clientToken: String?
+
+    var redirectManager: PrimerHeadlessUniversalCheckout.NativeUIManager?
+    var logs: [String] = []
+    var primerError: Error?
+    var checkoutData: PrimerCheckoutData?
+
+    private var paymentId: String?
+    private var activityIndicator: UIActivityIndicatorView?
+
+    class func instantiate(settings: PrimerSettings, clientSession: ClientSessionRequestBody?, clientToken: String?) -> MerchantHeadlessViewController {
+        let mcvc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "MerchantHeadlessViewController") as! MerchantHeadlessViewController
+        mcvc.settings = settings
+        mcvc.clientSession = clientSession
+        mcvc.clientToken = clientToken
+        return mcvc
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        PrimerHeadlessUniversalCheckout.current.delegate = self
+        PrimerHeadlessUniversalCheckout.current.uiDelegate = self
+    }
+    
+    // MARK: - HELPERS
+    
+    private func showLoadingOverlay() {
+        DispatchQueue.main.async {
+            if self.activityIndicator != nil { return }
+            
+            self.activityIndicator = UIActivityIndicatorView(frame: self.view.bounds)
+            self.view.addSubview(self.activityIndicator!)
+            self.activityIndicator?.backgroundColor = .black.withAlphaComponent(0.2)
+            self.activityIndicator?.color = .black
+            self.activityIndicator?.startAnimating()
+        }
+    }
+    
+    private func hideLoadingOverlay() {
+        DispatchQueue.main.async {
+            self.activityIndicator?.stopAnimating()
+            self.activityIndicator?.removeFromSuperview()
+            self.activityIndicator = nil
+        }
+    }
+    
+    @IBAction func onCheckoutButtonTap(_ sender: Any) {
+        
+        let vc = MerchantHeadlessCheckoutAvailablePaymentMethodsViewController.instantiate(settings: settings, clientSession: clientSession, clientToken: clientToken)
+        navigationController?.pushViewController(vc, animated: true)
+    }
+    
+    @IBAction func onVaultButtonTap(_ sender: Any) {
+        showLoadingOverlay()
+        if let clientToken = clientToken {
+            PrimerHeadlessUniversalCheckout.current.start(withClientToken: clientToken, settings: self.settings, completion: { (pms, err) in
+                let vaultManager = PrimerHeadlessUniversalCheckout.VaultManager()
+                try! vaultManager.configure()
```

Comment:
Remove `!` and push to the the result screen (`MerchantResultViewController`) with the proper logs, as this way people testing with the app will understand what went wrong.